### PR TITLE
CyberSource: Void Acepts nil values

### PR DESCRIPTION
**Pull request by gasb150**
Summary
-------------------------
Improve void request to handle cases where we
don't receive an authorization, (timeOuts cases)

[SER-1636](https://spreedly.atlassian.net/browse/SER-1636)

Unit Tests
----------------
Finished in 3.532173 seconds.
167 tests, 981 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 100% passed

Remote Tests
----------------
Finished in 127.168569 seconds.
143 tests, 730 assertions, 3 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications 97.9021% passed

Rubocop
----------------
808 files inspected, no offenses detected

### CODE REVIEW

File: lib/active_merchant/billing/gateways/cyber_source.rb [hunk 1]
```diff
@@ -488,9 +488,13 @@ def reference_is_a_check?(payment_method_or_reference)
         payment_method_or_reference.is_a?(String) && payment_method_or_reference.split(';')[7] == 'check'
       end
 
-      def build_void_request(identification, options)
-        order_id, request_id, request_token, action, money, currency = identification.split(';')
-        options[:order_id] = order_id
+      def build_void_request(identification, options, money = nil)
+        if identification.nil?
```
Comment (javierpedrozaing) [inline lib/active_merchant/billing/gateways/cyber_source.rb:492]: I am not sure about this validation, maybe we can use something like this.

```suggestion
        `order_id, request_id, request_token, action, money, currency = identification&.split(';')`
```

**Review by javierpedrozaing — COMMENTED** [review]
I left a small comment, LGTM.

File: lib/active_merchant/billing/gateways/cyber_source.rb [hunk 2]
```diff
@@ -221,8 +221,8 @@ def purchase(money, payment_method, options = {})
         end
       end
 
-      def void(identification, options = {})
-        commit(build_void_request(identification, options), :void, nil, options)
+      def void(identification, options = {}, money = nil)
```
Comment (Heavyblade) [inline lib/active_merchant/billing/gateways/cyber_source.rb:224]: ⚠️ Warning - ❓ Question:

Core doesn't send money value, so what's the point of setting a default value in a way that will never be filled.

**Review by Heavyblade — COMMENTED** [review]
@gasb150 left you some feedback

File: lib/active_merchant/billing/gateways/cyber_source.rb (hunk 2 above)
Comment (gasb150) [reply lib/active_merchant/billing/gateways/cyber_source.rb:224]: My idea is to modify the void method in the backend, 

As far as I tested, the money value is required, so I need the money value in void, as other methods (with partial void) handle it.

File: test/remote/gateways/remote_cyber_source_test.rb [hunk 3]
```diff
@@ -412,6 +412,15 @@ def test_purchase_and_void
     assert_successful_response(void)
   end
 
+  def test_void_with_no_authorization_value
+    merchant_transaction_id = 'testTransaction131' + SecureRandom.hex(3)
+
+    @gateway.authorize(@amount, @credit_card, @options.merge(merchant_transaction_id:))
```
Comment (naashton) [inline test/remote/gateways/remote_cyber_source_test.rb:418]: It looks like you are setting the key in the merge, but not giving it a value `.merge(merchant_transaction_id:))`

**Review by naashton — CHANGES_REQUESTED** [review]
Looks pretty good, but there's one small issue in the remote test.

Can you also add a unit test here and include a validation that inspects the request for the `nil` value for the `authRequestID`. This is a good candidate for that type of test.

File: test/remote/gateways/remote_cyber_source_test.rb (hunk 3 above)
Comment (naashton) [reply test/remote/gateways/remote_cyber_source_test.rb:418]: Ah, thanks for the explantation!

**Review by naashton — APPROVED** [review]
Looks good! Thanks!
